src/state/nftMarket/reducer.ts

```diff
@@ -73,47 +69,58 @@ export const fetchCollection = createAsyncThunk<Record<string, Collection>, stri
  * and the Subgraph (dynamic market data)
  * @param collectionAddress
  */
-export const fetchNftsFromCollections = createAsyncThunk<Record<string, PancakeBunnyNftWithTokens>, string>(
+export const fetchNftsFromCollections = createAsyncThunk(
   'nft/fetchNftsFromCollections',
-  async (collectionAddress) => {
-    const [nfts, nftsMarket] = await Promise.all([
-      getNftsFromCollectionApi(collectionAddress),
-      getNftsFromCollectionSg(collectionAddress),
-    ])
+  async (collectionAddress: string) => {
+    try {
+      const [nfts, nftsMarket] = await Promise.all([
+        getNftsFromCollectionApi(collectionAddress),
+        getNftsFromCollectionSg(collectionAddress),
+      ])
+
+      if (!nfts?.data) {
+        return []
+      }
 
-    // Separate market data by token id
-    const nftsMarketObj: Record<string, TokenMarketData> = nftsMarket.reduce(
-      (accum, nftMarketData) => ({ ...accum, [nftMarketData.tokenId]: { ...nftMarketData } }),
-      {},
-    )
-
-    return mapValues<ApiResponseCollectionTokens['data'], PancakeBunnyNftWithTokens>(nfts?.data, (nft, bunnyId) => {
-      const tokens = nft.tokens.reduce((accum: Record<number, TokenMarketData>, tokenId: number) => {
-        const token = nftsMarketObj[tokenId]
-        return { ...accum, [tokenId]: token }
-      }, {})
-
-      // Generating attributes field that is not returned by API but can be "faked" since objects are keyed with bunny id
-      const attributes =
-        collectionAddress === pancakeBunniesAddress
-          ? [
-              {
-                traitType: 'bunnyId',
-                value: bunnyId,
-                displayType: null,
-              },
-            ]
-          : null
-      return {
-        name: nft.name,
-        description: nft.description,
-        collectionName: nft.collection.name,
-        collectionAddress,
-        image: nft.image,
-        tokens,
-        attributes,
+      if (collectionAddress === pancakeBunniesAddress) {
```

Comment:
here's the most intense change - mapping happens a bit different cause we now return `NftToken[]` from here and not object with pancakebunnies